roles/scm/git/tasks/pull.yml

```diff
@@ -0,0 +1,31 @@
+---
+
+- name: Prepare SCM credentials
+  include_tasks: prep.yml
+
+- name: Create temporary directory for the repository
+  tempfile:
+    state: directory
+    prefix: "repo-"
+  register: result_scm_dir
+  changed_when: false
+  when:
+    - not repository.path is defined
+
+- name: Set directory for the repository
+  set_fact:
+    scm_dir: "{{ repository.path if repository.path is defined else result_scm_dir.path }}"
+
+- name: "Clone repository to temporary location"
+  git:
+    clone: true
+    repo: "{{ scm_repository }}"
+    dest: "{{ scm_dir }}"
+    force: "{{ repository.force | default(false) }}"
+    key_file: "{{ ssh_key if ssh_key is defined else omit }}"
+    remote: "{{ repository.remote | default('origin') }}"
+    update: "{{ repository.update | default('true') | bool }}"
+    version: "{{ repository.version | default('HEAD') }}"
```

Comment:
In general, I'm a fan of using `default(omit)` instead of defaulting to the same as the default in the module - unless there's a very specific reason why we do not want to follow the module defaults as it evolves